components/FourPicsTable.js

```diff
@@ -1,47 +1,70 @@
 import { StyledImage } from "./StyledImage";
 import { StyledA } from "./StyledLink";
 import styled from "styled-components";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
 
 export default function FourPicsTable() {
+  const router = useRouter();
+  const { searchID } = router.query;
+
+  const {
```

Comment:
At the moment, when the page first renders, `searchID` is `undefined`. That means we would end up making a request to `/api/ChooseVariation/undefined` which i think would return a 404 error. We can prevent that by either providing a url to useSwr or `null`. Something like this. 

``` js 
const {
  data: shirts,
  isLoading,
  error,
} = useSWR(searchID ? `/api/ChooseVariation/${searchID}` : null, fetcher);

``` 

We would then also need to make sure to return loading if there are no shirts yet (i.e on the first render)

``` js
if (isLoading || !shirts) {
  return <div>loading...</div>;
}
```

You can read more on the subject here 
[conditional fetching with useSWR](https://swr.vercel.app/docs/conditional-fetching)

Reply:
thank you very much, i kind of went crazy over the course of all these dynamic routes